yark/archiver/config.py

```diff
@@ -0,0 +1,93 @@
+"""Configuration for metadata/downloads"""
+
+from typing import Optional, Any
+from colorama import Fore
+from ..logger import VideoLogger
+from pathlib import Path
+
+
+class Config:
+    max_videos: Optional[int]
+    max_livestreams: Optional[int]
+    max_shorts: Optional[int]
+    skip_download: bool
+    skip_metadata: bool
+    comments: bool
+    format: Optional[str]
+    proxy: Optional[str]
+
+    def __init__(self) -> None:
+        self.max_videos = None
+        self.max_livestreams = None
+        self.max_shorts = None
+        self.skip_download = False
+        self.skip_metadata = False
+        self.comments = False
+        self.format = None
+        self.proxy = None
+
+    def submit(self):
+        """Submits configuration, this has the effect of normalising maximums to 0 properly"""
+        # Adjust remaining maximums if one is given
+        no_maximums = (
+            self.max_videos is None
+            and self.max_livestreams is None
+            and self.max_shorts is None
+        )
+        if not no_maximums:
+            if self.max_videos is None:
+                self.max_videos = 0
+            if self.max_livestreams is None:
+                self.max_livestreams = 0
+            if self.max_shorts is None:
+                self.max_shorts = 0
+
+        # If all are 0 as its equivalent to skipping download
+        if self.max_videos == 0 and self.max_livestreams == 0 and self.max_shorts == 0:
+            print(
+                Fore.YELLOW
+                + "Using the skip downloads option is recommended over setting maximums to 0"
+                + Fore.RESET
+            )
+            self.skip_download = True
+
+    def settings_dl(self, path: Path) -> dict[str, Any]:
```

Comment:
Make a new type alias for yt-dlp settings, typescipt syntax for this would be `type YtDlpSettings = dict[str, Any]`